src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.WorkerThread.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using Internal.Runtime.Augments;
+
+namespace System.Threading
+{
+    internal static partial class ClrThreadPool
+    {
+        internal static class WorkerThread
+        {
+            private static LowLevelLifoSemaphore s_semaphore = new LowLevelLifoSemaphore(0, int.MaxValue);
+
+            private const int TimeoutMs = 20 * 1000;
+            
+            private static void WorkerThreadStart()
+            {
+                // TODO: Event: Worker Thread Start event
+
+                while (true)
+                {
+                    // TODO: Event:  Worker thread wait event
+                    while (s_semaphore.Wait(TimeoutMs))
+                    {
+                        if(!ThreadPoolWorkQueue.Dispatch())
+                        {
+                            continue;
+                        }
+
+                        RuntimeThread.CurrentThread.Priority = ThreadPriority.Normal;
+                        CultureInfo.CurrentCulture = CultureInfo.InstalledUICulture;
+                        CultureInfo.CurrentUICulture = CultureInfo.InstalledUICulture;
```

Comment:
Also, after Dispatch returns, the original code is calling ShouldAdjustMaxWorkersActive and AdjustMaxWorkersActive. I don't think those are necessary because Dispatch calls NotifyWorkItemComplete, but please double-check if that is necessary for something.